File: algorithms/tree/tree.cpp
#include <iostream>
#include <vector>

struct Node {
  int value;
  std::vector<Node*> children; 

  void add_child(int value) {
    Node* node = new Node();
    node->value = value;
    children.push_back(node);
  }
};

int main() {
  Node root;
  root.value = 10;

  root.add_child(30);
  root.add_child(40);
  root.add_child(50);
  root.add_child(60);

  root.children[0]->add_child(50);

  printf("Number of children %d\n", root.children.size());
  printf("Number of children %d\n", root.children[1]->children.size());
  
}

File: algorithms/tree/tree.py
class Node:
  def __init__(self,value):
    self.value=value
    self.nodes=[]
  
  def add_child(self,value):
    self.nodes.append(Node(value))

root=Node(10)
root.add_child(15)
root.add_child(20)
root.add_child(35)

root.nodes[0].add_child(10)

print(len(root.nodes))
print(len(root.nodes[0].nodes))